jacoco-maven-plugin/src/org/jacoco/maven/ReportAggregateMojo.java

```diff
@@ -131,6 +138,10 @@ File getOutputDirectory() {
 	void createReport(final IReportGroupVisitor visitor,
 			final ReportSupport support) throws IOException {
 		final IReportGroupVisitor group = visitor.visitGroup(title);
+		if (includeCurrentProject) {
```

Comment:
May I propose to extract the following method:


	private void processProject(final ReportSupport support,
			final IReportGroupVisitor group, final MavenProject project)
			throws IOException {
		support.processProject(group, project.getArtifactId(), project,
				getIncludes(), getExcludes(), sourceEncoding);
	}

This will avoid code duplication because we can simply call it here with

    processProject(support, group, project);

and the original call becomes

    processProject(support, group, dependency);

Reply:
Changed this per your suggestion